internal_transfer_with_agreed_amount/wizards/internal_transfer_multicurrency.py

```diff
@@ -37,13 +37,23 @@ def _validate_currencies_in_internal_transfer(self, payment):
             )
         )
 
-    def _prepare_values_from_lines(self, lines, currency):
+    def _prepare_values_from_lines(self, lines, currency=None, is_same_company_currency=False):
         line = lines[0]
+        if is_same_company_currency:
+            return {"debit": self.agreed_amount} if line.debit > 0 else {"credit": self.agreed_amount}
         vals = {"debit": line.debit} if line.amount_currency > 0 else {"credit": line.credit}
         sign = 1 if line.amount_currency > 0 else -1
         vals.update({"amount_currency": sign * self.agreed_amount, "currency_id": currency.id})
         return vals
 
+    def _preprocessing_payment_with_agreed_amount(self, payment):
+        """This method is to be inherited in case pre-processing is required on the payments."""
+        return
+
+    def _postprocessing_payment_with_agreed_amount(self, payment):
```

Comment:
Since this is a hook and everything here is for internal transfers, I think it should be named something like:

```suggestion
    def _postprocess_payments_hooks(self, payment):
```

Same below.

Reply:
Commit message and method names updated.